pkg/prometheus/prometheus.go

```diff
@@ -0,0 +1,80 @@
+package prometheus
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+const prometheusDefaultPort = "2112"
+
+var (
+	ApiCallsFailure = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "api_call_failures_total",
+		Help: "The total number of api calls that failed by the updater",
+	}, []string{"node", "errorType"})
+
+	OperationDelay = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "operation_delay",
+			Help: "Represent the latency of the update operation",
+		}, []string{"node"})
+)
+
+func extractErrorType(error string) string {
+	// TODO temporary until PodresourceAPI errors will be standardize
+	if strings.Contains(error, "GetAllocatableResources disabled") {
+		return "feature-gate"
+	}
+	return "unexpected"
+}
+
+func nodeName() string {
+	val, ok := os.LookupEnv("NODE_NAME")
+	if !ok {
+		return ""
+	}
+	return val
+}
+
+func UpdateApiCallsFailureMetric(err error) {
+	errorType := extractErrorType(err.Error())
+	ApiCallsFailure.With(prometheus.Labels{
+		"node":      nodeName(),
+		"errorType": errorType,
```

Comment:
I'm not sure labels use the `camelCase`, let's check the standard.

Reply:
According to standard:
> Prometheus metrics and label names are written in snake_case

Will fix